airflow/models/dag.py

```diff
@@ -1727,86 +1727,63 @@ def add_logger_if_needed(ti: TaskInstance):
     @provide_session
     def create_dagrun(
         self,
-        state: DagRunState,
         *,
-        triggered_by: DagRunTriggeredByType | None,
-        logical_date: datetime | None = None,
-        run_id: str | None = None,
-        start_date: datetime | None = None,
-        external_trigger: bool | None = False,
+        run_id: str,
+        logical_date: datetime,
+        data_interval: tuple[datetime, datetime],
```

Comment:
Is the type expected to be different from the one above?

Reply:
Yes, I intentionally removed the None cases and require explicit values to be passed in. Previous you can pass None for some of the values, and the function magically infer the correct run_id and/or dates from various other values.

This becomes very complicated when `logical_date` could actually intended to be None (instead of just a marker to be inferred implicitly), so I moved the inferring logic to where it’s actually needed instead (it is mostly only used in tests).

Reply:
then, we probably could make it `DataInterval` instead?

Reply:
sounds good